File: fix-duplicate.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'templates.ts');

const content = fs.readFileSync(filePath, 'utf-8');
const lines = content.split('\n');

console.log(`Total lines: ${lines.length}`);

let firstDrawerStart = -1;
let firstDrawerEnd = -1;
let secondDrawerStart = -1;

for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes('drawer: `"use client"')) {
        if (firstDrawerStart === -1) {
            firstDrawerStart = i;
            console.log(`Found first drawer at line ${i + 1}`);
        } else if (secondDrawerStart === -1) {
            secondDrawerStart = i;
            console.log(`Found second drawer at line ${i + 1}`);
            break;
        }
    }
}

if (firstDrawerStart !== -1) {
    for (let i = firstDrawerStart; i < Math.min(firstDrawerStart + 100, lines.length); i++) {
        if (lines[i].includes('export { Drawer, DrawerPortal, DrawerOverlay, DrawerTrigger, DrawerClose, DrawerContent')) {
            for (let j = i; j < Math.min(i + 5, lines.length); j++) {
                if (lines[j].includes('`,')) {
                    firstDrawerEnd = j + 1;
                    console.log(`Found end of first drawer at line ${j + 1}`);
                    break;
                }
            }
            break;
        }
    }
}

if (firstDrawerStart !== -1 && firstDrawerEnd !== -1) {
    console.log(`\nRemoving lines ${firstDrawerStart + 1} to ${firstDrawerEnd}`);

    const newLines = [
        ...lines.slice(0, firstDrawerStart),
        ...lines.slice(firstDrawerEnd)
    ];

    const newContent = newLines.join('\n');
    fs.writeFileSync(filePath, newContent, 'utf-8');

    console.log('✅ Successfully removed duplicate drawer definition!');
    console.log(`New file has ${new Lines.length} lines (removed ${firstDrawerEnd - firstDrawerStart} lines)`);
} else {
    console.log('❌ Could not locate drawer definitions');
    console.log(`firstDrawerStart: ${firstDrawerStart}, firstDrawerEnd: ${firstDrawerEnd}`);
}

File: fix_duplicate.py
import re

# Read the file
with open(r'c:\Users\youne\OneDrive\Desktop\sekerbura\src\templates.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# Split by lines and find the duplicate drawer sections
lines = content.split('\n')

# Find line indices for both drawer definitions
first_drawer_start = None
first_drawer_end = None
second_drawer_start = None

for i, line in enumerate(lines):
    if 'drawer: `"use client"' in line and first_drawer_start is None:
        first_drawer_start = i
    elif 'drawer: `"use client"' in line and first_drawer_start is not None:
        second_drawer_start = i
        break

# Find the end of first drawer (look for export { Drawer...)
if first_drawer_start:
    for i in range(first_drawer_start, min(first_drawer_start + 100, len(lines))):
        if 'export { Drawer, DrawerPortal, DrawerOverlay, DrawerTrigger, DrawerClose, DrawerContent' in lines[i]:
            first_drawer_end = i + 2  # Include the `, line after export
            break

print(f"First drawer: lines {first_drawer_start} to {first_drawer_end}")
print(f"Second drawer starts at: line {second_drawer_start}")

if first_drawer_start and first_drawer_end:
    # Remove the first drawer definition
    new_lines = lines[:first_drawer_start] + lines[first_drawer_end:]
    new_content = '\n'.join(new_lines)
    
    # Write back
    with open(r'c:\Users\youne\OneDrive\Desktop\sekerbura\src\templates.ts', 'w', encoding='utf-8') as f:
        f.write(new_content)
    
    print("✅ Successfully removed duplicate drawer definition!")
    print(f"Removed lines {first_drawer_start} to {first_drawer_end}")
else:
    print("❌ Could not find drawer definitions to remove")
